### PR TITLE
Add withNodeJSEnv, withNodeJSEnvUnix and nodeJSDefaultVersion

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Added a few steps to configure NVM and Node. This is similar to the `withGoEnv` and `goDefaultVersion` steps but for `Node`.

Therefore it's an opinionated approach to install node with `nvm` and relies in the `.nvmrc` file to be stored in the repository to use a different version, this is our way to embrace normalisation.

## Why is it important?

DRY

## Related issues

Notifies https://github.com/elastic/synthetics/pull/393

Tested in https://github.com/elastic/synthetics/pull/419

### CODE REVIEW

File: vars/README.md [hunk 1]
```diff
@@ -2225,6 +2225,13 @@ Return the architecture in the current worker using the labels as the source of
  def arch = nodeArch()
 ```
 
+## nodeDefaultVersion
+Return the value in the file `.nvmrc`, or a default value.
+
+```
+  goDefaultVersion()
```
Comment (mdelapenya) [inline vars/README.md:2232]:
```suggestion
  nodeDefaultVersion()
```

**Review by mdelapenya — APPROVED** [review]
LGTM, although I have the feeling that `withNode` recalls me the Jenkins Node entity.

Is it possible to rename these steps to append the `JS` suffix?
- `withNodeJSEnv`
- `withNodeJSEnvUnix`
- `nodeJSDefaultVersion` 

> Don't care if it's NodeJs or NodeJS 🤷

**Comment by v1v** [comment]
> LGTM, although I have the feeling that withNode recalls me the Jenkins Node entity.

Indeed, I somehow was getting the same feeling, `withNode` is in place so let me rename it with `nodejs` instead `node`, I'll use lowercase for simplicity. I hope they wont' get crazy if I skip the `.` from `Node.js`

**Comment by mdelapenya** [comment]
>  I hope they wont' get crazy if I skip the `.` from `Node.js`

I'd say that the `JS` is a de-facto standard for anything JS: https://www.w3schools.com/js/js_conventions.asp, so I'd bet `withNodeJS` would be more accurate. Apart from that, I'm ok with the changes

File: vars/withNodejsEnv.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Install Node.js with NVM and run some command in a pre-configured environment multiplatform. For such
+it's recommended to use the `cmd` step.
+
+```
+  withNodejsEnv(version: '14.17.5'){
+    cmd(label: 'Node version', script: 'node --version')
+  }
```
Comment (kuisathaverat) [inline vars/withNodejsEnv.txt:7]:
```suggestion
  withNodejsEnv(version: '14.17.5'){
    cmd(label: 'Node version', script: 'node --version')
  }
  
  withNodejsEnv(version: readFile(file: '.nvmrc').trim()){
    cmd(label: 'Node version', script: 'node --version')
  }
```

**Review by kuisathaverat — APPROVED** [review]

File: vars/withNodejsEnv.txt (hunk 2 above)
Comment (v1v) [reply vars/withNodejsEnv.txt:7]: `version` already supports the `readFile '.nvmrc'` see https://github.com/elastic/apm-pipeline-library/pull/1402/files#diff-425e2866e7902441a31cb5a9e3a92a73ea68ee28c100cd097b2ab70d7ae50dfbR25-R28

**Pull request merged**